### PR TITLE
[config-plugins] improve invalid config error message

### LINKED ISSUE #31601: PluginError when using plugins with parameters
### Minimal reproducible example

https://github.com/teisen/expolocationtest/blob/main/app.json

### What platform(s) does this occur on?

Android, iOS, Web

### Where did you reproduce the issue?

in a development build

### Summary

Hello. I am still new to Expo, but i was trying to add location to my app and run into the below problem. I am not able to add plugins that need a configuration in the app.json.
Besides the issue with the location plugin  also noticed that adding firebase plugiuns that need configuration cauised the same problem, so i assume the problem is not specific to the location plugin. 

Steps to reproduce: 
Created new, empty expo app with: npx install-expo-modules@latest

Install expo-location: npx expo install expo-location

Configuration in app.json/app.config.js:
    "plugins": [
      "expo-router",
      "expo-location",
        {
          "locationAlwaysAndWhenInUsePermission": "Allow $(PRODUCT_NAME) to use your location."
        }
    ],

Error:
C:\expolocationtest>npx expo start
Starting project at C:\expolocationtest
PluginError: Plugin is an unexpected type: object
PluginError: Plugin is an unexpected type: object
    at withStaticPlugin (C:\expolocationtest\node_modules\@expo\config-plugins\build\plugins\withStaticPlugin.js:130:11)
    at C:\expolocationtest\node_modules\@expo\config-plugins\build\plugins\withPlugins.js:30:84
    at Array.reduce (<anonymous>)
    at withPlugins (C:\expolocationtest\node_modules\@expo\config-plugins\build\plugins\withPlugins.js:30:18)
    at withConfigPlugins (C:\expolocationtest\node_modules\@expo\config\build\plugins\withConfigPlugins.js:35:47)
    at fillAndReturnConfig (C:\expolocationtest\node_modules\@expo\config\build\Config.js:229:78)
    at getConfig (C:\expolocationtest\node_modules\@expo\config\build\Config.js:286:10)
    at startAsync (C:\expolocationtest\node_modules\@expo\cli\build\src\start\startAsync.js:131:70)
    at expoStart (C:\expolocationtest\node_modules\@expo\cli\build\src\start\index.js:138:12)

Thanks!

### Environment

```text
expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 20.17.0 - C:\Program Files\nodejs\node.EXE
      npm: 10.8.2 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: AI-241.19072.14.2412.12360217
    npmPackages:
      expo: ~51.0.28 => 51.0.32
      expo-router: ~3.5.23 => 3.5.23
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.74.5 => 0.74.5
      react-native-web: ~0.19.10 => 0.19.12
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
PluginError: Plugin is an unexpected type: object
    at withStaticPlugin (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:425597)
    at C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:422939
    at Array.reduce (<anonymous>)
    at withPlugins (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:422884)
    at withConfigPlugins (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:464021)
    at fillAndReturnConfig (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:448476)
    at getConfig (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:449293)
    at actionAsync (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:567154)
    at async run (C:\Users\teise\AppData\Local\npm-cache\_npx\89957a0324271eeb\node_modules\expo-doctor\build\index.js:59:568832) {
  isPluginError: true,
  code: 'INVALID_PLUGIN_TYPE',
  cause: undefined
}

Node.js v20.17.0
```

### PR DESCRIPTION
**Pull request by vonovak**
# Why

closes #31601

the snapshot update (2nd commit) is unrelated to other changes

# How

makes the error message more helpful

# Test Plan

tests pass

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by vonovak** [comment]
* **#31936** <a href="https://app.graphite.dev/github/pr/expo/expo/31936?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vonovak and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/@expo/config-plugins/src/plugins/withStaticPlugin.ts [hunk 1]
```diff
@@ -128,8 +128,23 @@ export const withStaticPlugin: ConfigPlugin<{
...
+If you tried to provide parameters to a config plugin, make sure the plugin configuration is wrapped by square brackets. Ex:\n
+[
+  "some-config-plugin",
+  {
+    "someParam": "someValue"
+  }
+]
+\n
+Please consult the package documentation on how to correctly configure the plugin.`,
+        'INVALID_PLUGIN_TYPE'
```
Comment (byCedric) [inline packages/@expo/config-plugins/src/plugins/withStaticPlugin.ts:143]:
```suggestion
        `Plugin is an unexpected object with keys: "${Object.keys(pluginResolve).join(', ')}".\n
If you tried to provide parameters to a config plugin, make sure the plugin configuration is wrapped by square brackets. Ex:\n
[
  "some-config-plugin",
  {
    "someParam": "someValue"
  }
]
\n
See the package documentation on how to correctly configure the plugin.`,
        'INVALID_PLUGIN_TYPE'
```

File: packages/@expo/config-plugins/src/plugins/withStaticPlugin.ts [hunk 2]
```diff
@@ -128,8 +128,23 @@ export const withStaticPlugin: ConfigPlugin<{
...
+      );
+    }
     throw new PluginError(
-      `Plugin is an unexpected type: ${typeof pluginResolve}`,
+      `Plugin is an unexpected type: ${typeof pluginResolve}. Please consult the documentation on how to correctly configure a config plugin.`,
```
Comment (byCedric) [inline packages/@expo/config-plugins/src/plugins/withStaticPlugin.ts:147]:
```suggestion
      `Plugin is an unexpected type: ${typeof pluginResolve}. See the package documentation on how to correctly configure the plugin.`,
```

File: packages/@expo/config-plugins/src/plugins/__tests__/withStaticPlugin-test.ts [hunk 3]
```diff
@@ -44,10 +44,13 @@ describe(withStaticPlugin, () => {
...
-        // @ts-ignore -- invalid type
-        plugin: {},
+        plugin: {
+          // @ts-ignore -- invalid type
+          key: 'value',
+          key2: 'value2',
+        },
       })
-    ).toThrow('Plugin is an unexpected type: object');
+    ).toThrow('Plugin is an unexpected type. Saw an object with keys: "key, key2"');
```
Comment (byCedric) [inline packages/@expo/config-plugins/src/plugins/__tests__/withStaticPlugin-test.ts:53]:
```suggestion
    ).toThrow('Plugin is an unexpected object with keys: "key, key2"');
```

**Review by byCedric — APPROVED** [review]
I don't have a strong opinion on having multi-line `PluginError` message, let's see how that works in practice.

The error message itself:
- Avoid using "please" in errors, mostly to sound more confident
- Simplified the multi-line error's first line to closely match the normal `PluginError` message for non-objects
- I'm not sure if "consult" is the easiest for non-native English speakers (I asked claude), used "See" instead.

<details><summary>Claude's take on <code>consult</code></summary>

image


</details>

**Comment by vonovak** [comment]
merged because main has the same [failure](https://github.com/expo/expo/actions/runs/11477307243/job/31939090075)

**Pull request merged**